### PR TITLE
Features/pico3

### PR DESCRIPTION
**Pull request by mastazi**
Adding support for Erica Pico System III. At the moment, I'm not fully satisfied about click areas being not aligned with their respective images, also see comments in erica-pico-iii.js in relation to some constants where I'm not sure what the appropriate value should be.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -10,12 +10,12 @@
...
   <body>
     <main role="main" class="container">
-      <h1>Behringer Neutron</h1>
+      <h1>Erica Pico System III</h1>
 
       <form accept-charset="UTF-8" action="/patches" enctype="multipart/form-data" method="post">
       <input id="patch_data" name="patch[data]" type="hidden" value="{}">
         <section class="row row-front-panel">
-          <canvas id="front-panel" data-device="behringer-neutron" width="1598" height="530"></canvas>
+          <canvas id="front-panel" data-device="erica-pico-iii" width="1786" height="966"></canvas>
```
Comment (zwippie) [inline index.html:18]: Change this to have `width="1650" height="892"` and all hitbox issues are gone.

File: js/erica-pico-iii.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  {idx: 27, x: 1423, y: 215, type: "small", name: "MIX2 LVL 2"},
+  {idx: 28, x: 1423, y: 323, type: "small", name: "MIX2 LVL 3"},
+  {idx: 29, x: 1531, y: 215, type: "small", name: "MIX3 LVL 1"},
+  {idx: 30, x: 1531, y: 323, type: "small", name: "MIX3 LVL 2"},
+]
+
+// Not sure why radius has to be 90, but after some trial and error I found 
+// it works best in this case (hitboxes of different knobs won't overlap when I choose 90)
+const KNOB_TYPES = {
+  small: {radius: 90, limit: 280}
```
Comment (zwippie) [inline js/erica-pico-iii.js:97]: `small: {radius: 25, limit: 280}`

File: js/erica-pico-iii.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+]
+const TOGGLE_TYPES = {
+  horiztwo: {values: ["left", "right"]},
+  horizthree: {values: ["left", "centerhor", "right"]},
+  verttwo: {values: ["up", "down"]},
+  vertthree: {values: ["up", "center", "down"]}
+}
+
+const TOGGLE_WIDTH = 50 // This is not working in the way I would expect...
+const TOGGLE_HEIGHT = 0 // If I don't set this to 0, the toggle hitbox interferes with elements below the toggle...
```
Comment (zwippie) [inline js/erica-pico-iii.js:117]: Set both to `40` (a little over half the max width/height of a toggle switch).

File: js/erica-pico-iii.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      knobTypes: KNOB_TYPES,
+      toggles: TOGGLES,
+      toggleOptions: {
+        types: TOGGLE_TYPES,
+        width: TOGGLE_WIDTH,
+        height: TOGGLE_HEIGHT
+      },
+      initialValues: {
+        knobValues: initialKnobValues(),
+        toggleValues: Array(19).fill(0)
```
Comment (zwippie) [inline js/erica-pico-iii.js:145]: `Array(8).fill(0)`

**Review by zwippie — COMMENTED** [review]
Thanks, great work!

Fixed the hitbox issues you encountered. Should be working better now.

Some comments about the images:

1. The file size of the front panel image is a bit too large (2.06MB). Try to save it as a lower quality (8-bit?) png so it's about 500KB.
2. The `knob-small` image should have equal width and height. It also looks a bit slanted. Can you try to make a new knob-small image?

**Comment by mastazi** [comment]
After applying your suggestions it works great! Thank you! I'm going to submit a new pr

**Pull request closed without merging**